src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectAssetFileWatcher.cs

```diff
@@ -175,11 +183,21 @@ private void UnregisterFileWatcherIfAny()
             }
         }
 
-        protected override void Dispose(bool disposing)
+        protected override async Task DisposeCoreAsync(bool disposing)
         {
             if (disposing)
             {
-                _treeWatcher.Dispose();
+                _cancellationTokenSource.Cancel();
```

Comment:
You don't need to do that - Dispose will not be called before Init has finished. See the constructor of OnceInitializedOnceDisposedAsync.